app/admin/users.rb

```diff
@@ -0,0 +1,104 @@
+ActiveAdmin.register User do
+  after_create do |user|
+    # set role after creation
+    case params[:user][:role_type]
+    when 'Administrator'
+      user.role = Administrator.create
+    when 'Therapist'
+      user.role = Therapist.create
+    when 'Donor'
+      user.role = Donor.create
+    end
```

Comment:
Debating whether the user should be scrapped if the role provided isn't one of these three - currently it makes a user without a role, and there's no real way to populate them after the fact currently (since that shouldn't ever happen).